src/main/java/uk/gov/pay/ledger/report/dao/ReportDao.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.pay.ledger.report.dao;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.Query;
+import uk.gov.pay.ledger.report.params.PaymentsReportParams;
+import uk.gov.pay.ledger.transaction.model.TransactionType;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class ReportDao {
+    private static final String COUNT_TRANSACTIONS_BY_STATE = "SELECT state, count(1) FROM transaction t " +
+            "WHERE type = :transactionType::transaction_type " +
+            ":searchExtraFields " +
+            "GROUP BY state";
+
+    private final Jdbi jdbi;
+
+    @Inject
+    public ReportDao(Jdbi jdbi) {
+        this.jdbi = jdbi;
+    }
+
+    public List<PaymentCountByState> getPaymentCountsByState(PaymentsReportParams params) {
+        return jdbi.withHandle(handle -> {
+            String template = createSearchTemplate(params.getFilterTemplates(),
+                    COUNT_TRANSACTIONS_BY_STATE,
+                    true);
+
+            Query query = handle.createQuery(template)
+                    .bind("transactionType", TransactionType.PAYMENT);
+            params.getQueryMap().forEach(query::bind);
+
+            return query.map((rs, rowNum) -> {
+                String state = rs.getString("state");
+                Long count = rs.getLong(2);
```

Comment:
Could this be aliased as `count` in the query so that if further things are added to this it won't require the index to be correct?

Reply:
yes, good idea